### ISSUE TITLE
Check about lint warnings - JSX props should not use arrow functions

### ISSUE THREAD

**Issue by roienatan** [daostack/alchemy#1867]
We've around 35 warnings like this in Alchemy. Theoretically it's not good for the performance since when passing an arrow function as a prop, the function is recreated on each render.

**Comment by dkent600** [comment]
Should be done in 1.0 and merged up to 2.0

**Comment by roienatan** [comment]
I did some investigation regarding this warning. I’ll start from the **conclusion**.
**Solving this is inconclusive and depends on a specific case.**
**Explanation:** 
Ideally when passing a function to a React component it’s better to pass a **reference** of the function instead of the function itself. 
If you pass the function itself, every time the component updates a new function is created and it can affect the performance, while passing a reference of the function is more effective.  
However, optimization is not free and usually we’ll need to compensate in another place. If we pass callback that does not accept parameters, this is the easy scenario to solve, we can just define it above and pass the function by reference like this: 
`onChange={ handleChange }` or `onChange={ this.handleChange }`
Things start to get more complicated when the callback accepts params – now we must write something like this:
`onChange={ param => handleChange(param) }`

This is not the reference of the function anymore and the `linter` throws a warning. Solving this is more complicated and in this case, we must change a little bit the logic and apply the _“separation of concerns”_ design principle. That means more code, not always possible and sometimes we might even end with worst performance. Another solution is to use `useCallback` from the new React Hooks but again, changing the logic is required here unless we write new components from screch and we can think about the logic before writing. 

That’s why each case should be investigated specifically. For example, if we know that a component will not get rendered many times, it’s better to pass the callback itself since it’s much easier to write, less code and much more readable. 

As I saw, most of our 35 warnings in Alchemy are from the `complicated` type that I explained above, yet I saw few cases that can be optimized easily.

Will update soon...
